### PR TITLE
Fixes enum comparisons

### LINKED ISSUE #2881: [Conditionals] State generator state cannot be evaluated
Configuration is being saved as string but data is number. Also, it appears telemetry name is trying to be cast as a Number.

state_generator_output

state_generator_name

### PR DESCRIPTION
**Pull request by shefalijoshi**
Adds check for undefined

Resolves #2881

Also checks for undefined for is one of and is not one of operations.

**Author Checklist**

- Changes address original issue? Yes
- Unit tests included and/or updated with changes? Yes
- Command line build passes? Yes
- Changes have been smoke-tested? Yes

### CODE REVIEW

File: src/plugins/condition/utils/operations.js [hunk 1]
```diff
@@ -226,7 +228,9 @@ export const OPERATIONS = [
     {
         name: 'enumValueIsNot',
         operation: function (input) {
-            return input[0] !== input[1];
+            let stringInputs = [];
+            input.forEach(inputValue => stringInputs.push(inputValue !== undefined ? inputValue.toString() : ''));
```
Comment (nikhilmandlik) [inline src/plugins/condition/utils/operations.js:232]: may be create new function, as it is used in other place as well.

File: src/plugins/condition/utils/operationsSpec.js [hunk 2]
```diff
@@ -87,4 +89,54 @@ describe('Is one of and is not one of operations', function () {
...
+
+    it('should evaluate enumValueIs to false for undefined input', () => {
+        const inputs = [undefined, "45"];
+        expect(!!enumIsOperation.operation(inputs)).toBeFalse();
+    });
+
+    it('should evaluate enumValueIsNot to true for undefined input', () => {
+        const inputs = [undefined, "45"];
+        expect(!!enumIsNotOperation.operation(inputs)).toBeTrue();
+    });
```
Comment (nikhilmandlik) [inline src/plugins/condition/utils/operationsSpec.js:141]: Nice variety 👍

**Comment by nikhilmandlik** [comment]
Small change else LGTM

File: src/plugins/condition/utils/operations.js (hunk 1 above)
Comment (shefalijoshi) [reply src/plugins/condition/utils/operations.js:232]: Good point. Will do.

**Comment by shefalijoshi** [comment]
Merging as this is approved.

**Pull request merged**